components/amara/amara.app.mjs

```diff
@@ -0,0 +1,589 @@
+        params: {
+          format,
+        },
+      });
+    },
+    async deleteSubtitles({
+      $, videoId, language,
+    }) {
+      return this._makeRequest({
+        $,
+        method: "delete",
+        path: `/videos/${videoId}/languages/${language}/subtitles`,
+      });
+    },
+    async listActions({
+      $, videoId, language,
+    }) {
+      return this._makeRequest({
+        $,
+        path: `/videos/${videoId}/languages/${language}/subtitles/actions`,
+      });
+    },
+    async performAction({
+      $, videoId, language, action,
+    }) {
+      return this._makeRequest({
+        $,
+        method: "post",
+        path: `/videos/${videoId}/languages/${language}/subtitles/actions`,
+        data: {
+          action,
+        },
+      });
+    },
+    async getTeam({
+      $, teamId,
+    }) {
+      return this._makeRequest({
+        $,
+        path: `/teams/${teamId}`,
+      });
+    },
+    async getTeams({
+      $, url, params,
+    }) {
+      return this._makeRequest({
+        $,
+        url,
+        path: "/teams",
+        params,
+      });
+    },
+    async getTeamProjects({
+      $, url, team, limit,
+    }) {
+      return this._makeRequest({
+        $,
+        url,
+        path: `/teams/${team}/projects`,
+        params: {
+          limit,
+        },
+      });
+    },
+    async getLanguages({ $ } = {}) {
+      return this._makeRequest({
+        $,
+        path: "/languages",
+      });
+    },
+    async getTeamNotifications({
+      $, url, team, params,
+    }) {
+      return this._makeRequest({
+        $,
+        url,
+        path: `/teams/${team}/notifications`,
+        params,
+      });
+    },
+    async getTeamActivity({
+      $, url, team, params,
+    }) {
+      return this._makeRequest({
+        $,
+        url,
+        path: `/teams/${team}/activity`,
+        params,
+      });
+    },
+    /**
+     * getResourcesStream always gets the latest resources from the API.
+     * @param {Object} args - all arguments to pass to the `getResourcesStream` function
+     * @param {Function} args.resouceFn - the name of the resource function to call
+     * @param {Object} args.resourceFnArgs - the arguments object to pass to the resource function
+     * @param {number} [args.offset] - the offset to start at
+     * @param {number} [args.limit=100] - the number of resources to get per page
+     * @param {number} [args.max=300] - the maximum number of resources to get
+     * @param {string} [args.lastResourceStr] - the last resource string in cache
+     * to validate against. This parameter is only passed in from sources.
+     * @returns {Iterable} - Iterable that yields resources,
+     *  the first element is the last resource string in cache
+     */
+    async *getResourcesStream({
+      resourceFn,
+      resourceFnArgs,
+      offset,
+      limit = constants.DEFAULT_PAGE_LIMIT,
+      max = constants.DEFAULT_MAX_ITEMS,
```

Comment:
If there are more than `DEFAULT_MAX_ITEMS` (`300`) new events between runs of an event source, I think only the first 300 will be processed, and the rest will be missed. Should the default value for the `max` arg be removed?
```js
      max,
```
And change the `if` statement, https://github.com/PipedreamHQ/pipedream/blob/e9d0aac9692961d36a943f4d7e16fd770e404856/components/amara/amara.app.mjs#L583
to?:
```js
        if (!nextResponse?.meta.next || (max && resourcesCount >= max)) {
```